node/services/service.go

```diff
@@ -132,20 +136,30 @@ func ShareService(lc fx.Lifecycle, bServ blockservice.BlockService, avail share.
 
 // DASer constructs a new Data Availability Sampler.
 func DASer(
+	ctx context.Context,
 	lc fx.Lifecycle,
 	avail share.Availability,
 	sub header.Subscriber,
 	hstore header.Store,
 	ds datastore.Batching,
+	fService fraud.Service,
 ) *das.DASer {
-	das := das.NewDASer(avail, sub, hstore, ds)
+	das := das.NewDASer(avail, sub, hstore, ds, fService)
 	lc.Append(fx.Hook{
 		OnStart: das.Start,
 		OnStop:  das.Stop,
 	})
+	go fraud.OnBEFP(fxutil.WithLifecycle(ctx, lc), fService, das.Stop)
 	return das
 }
 
+// FraudService constructs fraud proof service
+func FraudService(sub *pubsub.PubSub, hstore header.Store) (fraud.Service, fraud.Service) {
+	f := fraud.NewService(sub, hstore.GetByHeight)
+	f.RegisterUnmarshaler(fraud.BadEncoding, fraud.UnmarshalBEFP) //nolint:errcheck
```

Comment:
Why no error check?

Reply:
Though it's not needed if we are registering on system start

Reply:
Added